server/src/test/java/org/elasticsearch/index/mapper/DynamicMappingTests.java

```diff
@@ -607,7 +607,7 @@ public void testDateDetectionInheritsFormat() throws Exception {
                         .endObject()
                     .endObject()
                     .startObject()
-                        .startObject("dates")
+                        .startObject("dates_with_explicit_format")
```

Comment:
After having to change this test I'm not entirely sure anymore about the approach, in particular whether this test was lazily "missusing" a leniency in our code by using the name twice or wether this should work. In the end both templates seem to match different fields, so I'm not sure why the test used the same name?